Hannah-Sten/TeXiFy-IDEA
# Fix #2681: add parent sections to structure view after orphaned sections

Fix #2681

#### Summary of additions and changes

* Structure view elements were only added if the structure of all sections was top to bottom. So `section - subsection - section` was fine, but for `subsection - section - subsection` the `section` would not show up in the structure view.

#### How to test this pull request

Check the structure view for the following file:

```latex
\chapter{Zero}

\section{fd}

\part{First part}

\subsubsection{fd}

\section{a}

\chapter{One}

\section{B}

\chapter{Two}

\part{Second part}

\chapter{Three}
```
